proposals/0075-precompile-for-secp256r1-sigverify.md

```diff
@@ -0,0 +1,298 @@
+message by simply flipping the `s` value over the curve.
+
+As the primary goal of the `secp256r1` program is secure signature validation
+for authentication purposes, the precompile must mitigate these attacks
+by enforcing the usage of `lowS` values, in which `s <= n/2`.
+
+As such, the program must immediately fail upon the detection of any
+signature that includes a `highS` value. This prevents any accidental
+succeptibility to signature malleability attacks.
+
+Note: The existing `secp256k1` precompile makes no attempt attempt to mitigate
+s malleability, as doing so would go against its primary goal of achieving
+`ecrecover` parity with EVM.
+
+### Implementation
+
+### Program
+
+ID: `Secp256r1SigVerify1111111111111111111111111`
+
+The program instruction must be composed of the following struct:
+
+In Pseudocode:
+
+```
+struct Secp256r1SigVerifyInstruction {
+    count: uint8 LE,                          // Number of signatures to check
+    padding: uint8 LE,                        // Single byte padding
+    signatureOffsets: Array<Secp256r1SignatureOffsets>, // Array of structs
+}
+
+struct Secp256r1SignatureOffsets {
+    signature_offset: uint16 LE,              // Offset to signature
+    signature_instruction_index: uint8 LE,   // Instruction index to signature
+    public_key_offset: uint16 LE,             // Offset to public key
+    public_key_instruction_index: uint8 LE,  // Instruction index to  public key
+    message_data_offset: uint16 LE,           // Offset to start of message data
+    message_data_size: uint16 LE,             // Size of message data
+    message_instruction_index: uint8 LE,     // Instruction index to message
+}
+```
+
+Multiple signatures can be verified. If any of the signatures fail to verify,
+an error must be returned.
+
+If the instruction data is empty, the program must return an error.
+
+If `count == 0` and the length of the instruction data > 1, the program must
+return an error.
+
+If `count == 0` and the length of the instruction data is not at least
+`count * Secp256r1SignatureOffsets + 1`, the program must return an error.
+
+The instruction processing logic must follow the pseudocode below:
+
+```
+/// `data` is the secp256r1 program's instruction data. `instruction_datas` is
+/// the full slice of instruction datas for all instructions in the transaction,
+/// including the secp256r1 program's instruction data.
+
+/// length_of_data is the length of `data`
+
+/// SERIALIZED_OFFSET_STRUCT_SIZE is the length of the serialized
+/// Secp256r1SignatureOffsets struct
+
+if length_of_data == 0 {
+  return Error
+}
+count = data[0]
+if count == 0 && length_of_data > 1 {
+  return Error
+}
+if length_of_data < (count * SERIALIZED_OFFSET_STRUCT_SIZE + 1) {
+  return Error
+}
+instructions = instruction_datas;
+for i in 0..count {
+    signature = instructions[signature_instruction_index].data[signature_offset..signature_offset+64]
```

Comment:
You should add checks for indices and offsets being outside the range to the pseudocode